kaldb/src/main/java/com/slack/kaldb/logstore/opensearch/OpenSearchAdapter.java

```diff
@@ -174,7 +178,7 @@ public Query buildQuery(
         boolQueryBuilder.must(queryStringQueryBuilder);
       }
 
-      return boolQueryBuilder.toQuery(queryShardContext);
+      return boolQueryBuilder.rewrite(queryShardContext).toQuery(queryShardContext);
```

Comment:
should we add a unit test that fails without this fix?

Reply:
As discussed on Slack only some queries are throwing an exception around requiring rewriting, but I haven't been able to consistently identify which. I believe this is related to a change in #524 however, as after reverting that we haven't seen any more exceptions.

Regardless, rewriting this is a good idea as it allows for optimizing the query prior to execution.